advanced/Scripts/list.sh

```diff
@@ -5,261 +5,187 @@
 #
-# Whitelist and blacklist domains
+# allowlist and denylist domains
 #
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
-# Globals
-piholeDir="/etc/pihole"
-GRAVITYDB="${piholeDir}/gravity.db"
-# Source pihole-FTL from install script
-pihole_FTL="${piholeDir}/pihole-FTL.conf"
-if [[ -f "${pihole_FTL}" ]]; then
-    source "${pihole_FTL}"
+readonly PI_HOLE_SCRIPT_DIR="/opt/pihole"
+readonly utilsfile="${PI_HOLE_SCRIPT_DIR}/utils.sh"
+source "${utilsfile}"
+
+readonly apifile="${PI_HOLE_SCRIPT_DIR}/api.sh"
+source "${apifile}"
+
+# Determine database location
+DBFILE=$(getFTLConfigValue "files.database")
+if [ -z "$DBFILE" ]; then
+    DBFILE="/etc/pihole/pihole-FTL.db"
 fi
 
-# Set this only after sourcing pihole-FTL.conf as the gravity database path may
-# have changed
-gravityDBfile="${GRAVITYDB}"
+# Determine gravity database location
+GRAVITYDB=$(getFTLConfigValue "files.gravity")
+if [ -z "$GRAVITYDB" ]; then
+    GRAVITYDB="/etc/pihole/gravity.db"
+fi
 
-noReloadRequested=false
 addmode=true
 verbose=true
 wildcard=false
-web=false
 
 domList=()
 
 typeId=""
 comment=""
-declare -i domaincount
-domaincount=0
-reload=false
 
 colfile="/opt/pihole/COL_TABLE"
 source ${colfile}
 
-# IDs are hard-wired to domain interpretation in the gravity database scheme
-# Clients (including FTL) will read them through the corresponding views
-readonly whitelist="0"
-readonly blacklist="1"
-readonly regex_whitelist="2"
-readonly regex_blacklist="3"
-
-GetListnameFromTypeId() {
-    if [[ "$1" == "${whitelist}" ]]; then
-        echo "whitelist"
-    elif  [[ "$1" == "${blacklist}" ]]; then
-        echo "blacklist"
-    elif  [[ "$1" == "${regex_whitelist}" ]]; then
-        echo "regex whitelist"
-    elif  [[ "$1" == "${regex_blacklist}" ]]; then
-        echo "regex blacklist"
-    fi
-}
-
-GetListParamFromTypeId() {
-    if [[ "${typeId}" == "${whitelist}" ]]; then
-        echo "w"
-    elif  [[ "${typeId}" == "${blacklist}" ]]; then
-        echo "b"
-    elif  [[ "${typeId}" == "${regex_whitelist}" && "${wildcard}" == true ]]; then
-        echo "-white-wild"
-    elif  [[ "${typeId}" == "${regex_whitelist}" ]]; then
-        echo "-white-regex"
-    elif  [[ "${typeId}" == "${regex_blacklist}" && "${wildcard}" == true ]]; then
-        echo "-wild"
-    elif  [[ "${typeId}" == "${regex_blacklist}" ]]; then
-        echo "-regex"
-    fi
-}
-
 helpFunc() {
-    local listname param
-
-    listname="$(GetListnameFromTypeId "${typeId}")"
-    param="$(GetListParamFromTypeId)"
-
-    echo "Usage: pihole -${param} [options] <domain> <domain2 ...>
-Example: 'pihole -${param} site.com', or 'pihole -${param} site1.com site2.com'
-${listname^} one or more domains
+    echo "Usage: pihole ${abbrv} [options] <domain> <domain2 ...>
+Example: 'pihole ${abbrv} site.com', or 'pihole ${abbrv} site1.com site2.com'
+${typeId^} one or more ${kindId} domains
```

Comment:
Some additional needed here to line up with other changes to function names, current output of `pihole --help`:

```
54af332e58c9:/etc/pihole# pihole --help
Usage: pihole [options]
Example: 'pihole -w -h'
Add '-h' after specific commands for more information on usage

Whitelist/Blacklist Options:
  -w, whitelist       Whitelist domain(s)
  -b, blacklist       Blacklist domain(s)
  --regex, regex      Regex blacklist domains(s)
  --white-regex       Regex whitelist domains(s)
  --wild, wildcard    Wildcard blacklist domain(s)
  --white-wild        Wildcard whitelist domain(s)
                        Add '-h' for more info on whitelist/blacklist usage
```

Reply:
About the wording:

I'm not against the use of `denylist`, but the option is `-b` and the code also accepts `blocklist`. We already use `-d` for other option.

My questions are: 
- should we accept `denylist`, `blocklist` or both?
- if accepting both, shouldn't we show `blocklist` in help too?

Reply:
Let me first tell you that I added `blocklist` only to have a "justification" to keep the `-b` shortcut. And that only because `-d` is already taken. After having thought about this, I fully agree this wasn't a very good decision so I'm asking if we should rather:

1. Remove `-d` as shortcut for `debug` and add it to `denylist`
2. Remove `-a` for allowed and add semi-short keys like
   - `pihole -deny stuff.com`
   - `pihole -allow other.net`

   so we can keep `-d` for `debug`

Maybe no. 2 without `-` even, like
```
pihole allow example.org
```
which is basically natural language computing like.

Reply:
> Maybe no. 2 without `-` even, like

I like this

Reply:
Thanks, I went a bit further and allowed `not` for removals as well (`-d` and `--delmode` are still there) so you can undo
```
pihole allow example .com
```
by
```
pihole allow not example.com
```

I know we could have done something like `disallow` or even "`unallow`" but I didn't want to get too fancy because this would have needed more code changes.

Reply:
I like the solution.

Reply:
I'm not entirely convinced by `not`, but I'll not die on that hill

Reply:
We're still here and open for better alternatives

Reply:
Reading the comments again, I think the "allow not" and "deny not" will be confusing.

My suggestions is to use one of "remove", "delete" or "-d" or "-rm" to follow the natural language computing idea.

Examples:
```
pihole allow example.com             ->  add "example.com" to allowlist
pihole allowlist example.com         ->  add "example.com" to allowlist

pihole allowlist remove example.com  ->  remove "example.com" from allowlist
pihole allowlist delete example.com  ->  remove "example.com" from allowlist
pihole allowlist -d example.com      ->  remove "example.com" from allowlist
```

What do you think?

Reply:
Yes, I like it and committed this change right now.